### DISCUSSION TITLE
bun + aedes + websocket doesn't work?

### DISCUSSION THREAD

**Discussion by tsalgia** [moscajs/aedes#948] [Q&A] [answered]
I am developing with bun [www.bun.sh]  (instead of node.js) and was able to use aedes with the Mqtt server working
However, I was not able to get a websocket connection working
I copied and pasted the code from the MQTT server over WebSocket aedes example https://github.com/moscajs/aedes/blob/main/docs/Examples.md#mqtt-server-over-websocket 

I used MQTT Explorer to connect to this. However, the client disconnects after a few minutes (timeout waiting on puback from aedes).
I also tried using mqtt.js client and it behaves similarly. 
I did see this error in aedes.on('publish') - 
>client error Invalid header flag bits, must be 0x0 for puback packet

When I use mqtt protocol , everything works as expected.

When I switch away from bun to node, the websocket code works like a dream as well.

Would appreciate any ideas of how to resolve this issue or get to the root cause?
I will be cross-posting in the bun forum as well. 
(FYI - bun does support web sockets as a protocol in general)

**Comment by robertsLando** [comment]
@tsalgia Unfortunally I never used Bun so I cannot help you, anyway if you find a solution please post it here as it could happen to someone else

**Comment by mcollina** [comment, marked as answer]
Thanks for reporting.

Ultimately Bun does not implement all Node.js APIs. It's not a clean drop-in replacement for Node.js (despite their claims). Please refer to the Bun [Discord](https://bun.sh/discord) or [Issue Tracker](https://github.com/oven-sh/bun) for such questions.
